dot/network/block_announce.go

```diff
@@ -223,6 +232,10 @@ func (s *Service) validateBlockAnnounceHandshake(from peer.ID, hs Handshake) err
 // if some more blocks are required to sync the announced block, the node will open a sync stream
 // with its peer and send a BlockRequest message
 func (s *Service) handleBlockAnnounceMessage(from peer.ID, msg NotificationsMessage) (propagate bool, err error) {
+	if msg.IsHandshake() {
+		return false, errExpectedBlockAnnounceMsg
+	}
+
```

Comment:
This will always be false since `*BlockAnnounceMessage` always return `false`, given the type assertion right below.
Also why is this function not taking a `*BlockAnnounceMessage` directly instead of `NotificationsMessage`? The type assertion should be done at the calling layer imo, given the function name.

Reply:
that's cause it's a notification message handler. It has to follow certain signature of a message handler to be accepted by `RegisterNotificationsProtocol`.

Take a look at dot/network/service.go line 245

Reply:
> that's cause it's a notification message handler. It has to follow certain signature of a message handler to be accepted by

Why not change the `NotificationsMessageHandler` function argument? I don't see the point to have it as an interface at all. The caller `createNotificationsMessageHandler` should do the type assertion of the NotificationsMessage and pass it down as `*BlockAnnounceMessage`

> The type assertion below could fail.

Yes that's my point. If the type assertion succeeds, `IsHandshake()` is always false. So no point doing both checks, the existing type assertion is enough.

Reply:
addressed